src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -670,20 +691,73 @@ private function addService($id, Definition $definition)
      */
     private function addServices()
     {
+        $hasProxyCandidate = false;
+        $proxyDumper = $this->getProxyDumper();
         $publicServices = $privateServices = '';
         $definitions = $this->container->getDefinitions();
         ksort($definitions);
         foreach ($definitions as $id => $definition) {
+            $hasProxyCandidate = $hasProxyCandidate || $proxyDumper->isProxyCandidate($definition);
+            if ($definition->isSynthetic() || ($this->asFiles && $definition->isShared())) {
+                continue;
+            }
             if ($definition->isPublic()) {
                 $publicServices .= $this->addService($id, $definition);
             } else {
                 $privateServices .= $this->addService($id, $definition);
             }
         }
 
+        if ($this->asFiles) {
+            $publicServices = <<<EOF
+
+    /*{$this->docStar}
+     * {@inheritdoc}
+     */
+    protected function load(\$file, \$lazyLoad = true)
```

Comment:
even if the implementation in the parent class is exactly the same, this is needed to have access to private properties of *$this* class in required files.